build/Rulesets/Roslyn_BuildRules.ruleset

```diff
@@ -35,16 +36,28 @@
     <Rule Id="CA1720" Action="None" />
     <Rule Id="CA1721" Action="None" />
     <Rule Id="CA1724" Action="None" />
+    <Rule Id="CA1801" Action="None" />
+    <Rule Id="CA1802" Action="None" /> <!-- disabled due to lack of support for string literals -->
+    <Rule Id="CA1804" Action="None" />
+    <Rule Id="CA1806" Action="None" />
+    <Rule Id="CA1812" Action="None" /> <!-- disabled as we create many internal types via reflection-based mechanisms -->
+    <Rule Id="CA1814" Action="None" />
     <Rule Id="CA1815" Action="None" />
+    <Rule Id="CA1821" Action="None" />
+    <Rule Id="CA1823" Action="None" /> <!-- disabled due to many known bugs -->
```

Comment:
The main ones were unused field/parameters didn't seem to be respecting interpolated strings.